mqtt/mqtt-broker/src/persist.rs

```diff
@@ -378,13 +397,22 @@ where
                     path.clone(),
                 ))))
             });
-            let file = OpenOptions::new()
+            let mut file = OpenOptions::new()
                 .create(true)
                 .write(true)
                 .open(&path)
                 .context(ErrorKind::Persist(ErrorReason::FileOpen(path.clone())))?;
             debug!("{} opened.", path.display());
 
+            let version = bincode::serialize("Version 0.0.1")
```

Comment:
Seems a bit overkill I thought we can wrap `ConsolidatedState` into some struct/enum and just ser/de it:
```rust 
#[derive(Serialize, Deserialize)]
#[serde(tag = "version")]
enum FileFormat {
  #[serde(rename = 1)]
  V1(ConsolidatedState)
}
```
So no manual work involved, everything is done by serde

Reply:
+1 on this

Reply:
the problem is that the version has to be read before the processing begins on the state. Otherwise the persistor won't know which read function to call. 

For example, say we have 2 formats: simple binary and gzipped binary. The version information has to be separate from the state otherwise we won't know if it should be decompressed or not.